js_modules/dagster-ui/packages/ui-core/src/runs/AssetTagCollections.tsx

```diff
@@ -149,10 +157,20 @@ export function useAdjustChildVisibilityToFill(moreLabelFn: (count: number) => s
 
 export const AssetKeyTagCollection = React.memo((props: AssetKeyTagCollectionProps) => {
   const {assetKeys, useTags, maxRows, dialogTitle = 'Assets in run'} = props;
-  const {setShowMore, dialog} = useShowMoreDialog(dialogTitle, assetKeys, renderItemAssetKey);
 
   const count = assetKeys?.length ?? 0;
   const rendered = maxRows ? 10 : count === 1 ? 1 : 0;
+
+  const {sortedAssetKeys, slicedSortedAssetKeys} = React.useMemo(() => {
+    const sortedAssetKeys = assetKeys?.slice().sort(sortItemAssetKey) ?? null;
+    return {
+      sortedAssetKeys,
+      slicedSortedAssetKeys: sortedAssetKeys?.slice(0, rendered) ?? [],
```

Comment:
you shouldn't need a separate `slicedSortedAssetKeys`

Reply:
Just adding to this - I think it'd be ok to export just sortedAssetKeys here and still do a `sortedAssetKeys.slice(0, rendered)` down on line 183. Re-building the slice on each render (it won't be memoized anymore) isn't a big deal performance wise

Reply:
>isn't a big deal performance wise

I don't agree with this. The issue is that as the app evolves someone might add a component deeper in the tree that is sensitive to re-renders so changing references on every render is like leaving a footgun when this happens.

Reply:
We need both the sortedAssetKeys and slicedSortedAssetKeys (discussed in huddle) so the current implementation is fine